Shopify/checkout-sheet-kit-swift
# Generate entitlements as a build step

### What changes are you making?

Following https://github.com/Shopify/checkout-sheet-kit-swift/pull/239 work to add a script to generate the entitlements file, this PR aims to add a build phase step to automate the process.

- Consumers dont have to do a manual step in order to run the app
- Prevents compilation if you haven't populated your Storefront.xcconfig (which would be a runtime error otherwise)

### Status

This is currently sat in draft whilst I consider how to fix an issue, the script runs after the files are analysed in the prior build step, meaning that if the file doesn't exist it will fail the build, we want to generate prior to this or leave a stubbed file there, without git tracking being an issue during development.

### How to test

If you run the following commands from the root of the repo first
```shell
rm /Samples/MobileBuyIntegration/MobileBuyIntegration/MobileBuyIntegration.entitlements
```

Then kick off a build in xcode - you should observe that the build succeeds 
Things to note that should be present, the first run you should see something similar to:
```
Success: Entitlements file generated at MobileBuyIntegration/MobileBuyIntegration.entitlements with domain tiago-santoss-store.myshopify.com
```

<img width="1524" alt="image" src="https://github.com/user-attachments/assets/ef075f01-ca1e-4e02-865e-ef91fdb4c292">


Subsequent builds will indicate the fact it doesn't regenerate the file
<img width="687" alt="image" src="https://github.com/user-attachments/assets/108f8e1f-7054-42bd-86c8-a62933ba42c1">

---

### Before you merge

> [!IMPORTANT]
>
> - [ ] I've added tests to support my implementation
> - [ x ] I have read and agree with the [Contribution Guidelines](https://github.com/shopify/checkout-sheet-kit-swift/blob/main/.github/CONTRIBUTING.md).
> - [ x ] I have read and agree with the [Code of Conduct](https://github.com/shopify/checkout-sheet-kit-swift/blob/main/.github/CODE_OF_CONDUCT.md).
> - [ ] I've updated the [README](https://github.com/shopify/checkout-sheet-kit-swift).

---

<details>
<summary>Checklist for releasing a new version</summary>


</details>

> [!TIP]
> See the [Contributing documentation](https://github.com/shopify/checkout-sheet-kit-swift/blob/main/.github/CONTRIBUTING.md#releasing-a-new-version) for instructions on how to publish a new version of the library.
